crates/memory-testing/src/main.rs

```diff
@@ -0,0 +1,41 @@
+use std::{io::Read, str::FromStr};
+
+use bitwarden_crypto::{AsymmetricCryptoKey, SymmetricCryptoKey};
+
+fn main() {
+    let now = std::time::Instant::now();
+
+    let mut test_string = String::new();
+    test_string.push_str("THIS IS USED TO CHECK THAT ");
+    test_string.push_str("THE MEMORY IS DUMPED CORRECTLY");
+
+    // In HEX:
+    // KEY: 15f8 5554 ff1f 9852 1963 55a6 46cc cf99 1995 0b15 cd59 5709 7df3 eb6e 4cb0 4cfb
+    // MAC: 4136 481f 8581 93f8 3f6c 5468 b361 7acf 7dfb a3db 2a32 5aa3 3017 d885 e5a3 1085
+    let symm_key = SymmetricCryptoKey::from_str(
+        "FfhVVP8fmFIZY1WmRszPmRmVCxXNWVcJffPrbkywTPtBNkgfhYGT+D9sVGizYXrPffuj2yoyWqMwF9iF5aMQhQ==",
```

Comment:
Thoughts on proving the key as input from stdin? In the future I envision we would expand the tool with more scenarios and it might be beneficial to not burn in all variants of enc strings and crypto keys.

Reply:
Yeah reading from stdin or a json file sounds interesting, though I'd be worried about the std library or serde making unnecessary copies and affecting the results. I'll give it a try and see what happens.

Reply:
I pushed some new changes that migrate the scripts to rust binaries and loads the test cases through JSON, seems to be fine as long as you zeroize the string after parsing it to JSON.